stackhpc_monasca_agent_plugins/tests/unit/detection/test_ib_network.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2018 StackHPC Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+import mock
+
+import stackhpc_monasca_agent_plugins.detection.ib_network as ib_network
+
+
+class MockIBNetworkDetectPlugin(ib_network.IBNetworkDetect):
+    def __init__(self):
+        # Don't call the base class constructor
+        pass
+
+
+class TestIBNetworkDetect(unittest.TestCase):
+    def setUp(self):
+        self.ib_network = MockIBNetworkDetectPlugin()
+
+    def test_build_config(self):
+        config = self.ib_network.build_config()
+        self.assertIn('ib_network', config)
+
+    @mock.patch('os.path.isdir')
+    @mock.patch('pkgutil.find_loader')
+    def test__detect_ok(self, mock_isdir, mock_find_loader):
+        mock_isdir.return_value = True
+        mock_find_loader.return_value = 'Dummy loader'
+        self.ib_network._detect()
+        self.assertTrue(self.ib_network.available)
+
+    @mock.patch('os.path.isdir')
+    @mock.patch('pkgutil.find_loader')
+    def test__detect_no_plugin(self, mock_isdir, mock_find_loader):
+        mock_isdir.return_value = False
```

Comment:
Doesn't this need to be false to get to the plugin check? You can always use mock_find_loader.assert_called_once_with(...) to ensure it's been called.